### ISSUE TITLE
Trouble running locally the service, ui - `relation "public.flows_v3" does not exist`

### ISSUE THREAD

**Issue by gaborvecsei** [Netflix/metaflow-service#308]
I tried to follow the instructions in the `metaflow-ui` repo: https://github.com/Netflix/metaflow-ui/blob/master/docs/README.md to deploy a Metaflow locally (on my macbook).

From the docs:
```
# Set up metaflow service
$ git clone https://github.com/Netflix/metaflow-service.git && cd metaflow-service
# Running docker-compose.development.yml (recommended during development):
$ docker-compose -f docker-compose.development.yml up

# Run docker container using custom default API endpoint
$ docker run -p 3000:3000 -e METAFLOW_SERVICE=http://localhost:8083/ metaflow-ui:latest
```

But the service fails because of the following:

```
psycopg2.errors.UndefinedTable: relation "public.flows_v3" does not exist
```

Here you can find the whole trace from the start:

```
Creating metaflow-service_db_1 ... done
Creating metaflow-service_ui_backend_1 ... done
Creating metaflow-service_migration_1  ... done
Creating metaflow-service_metadata_1   ... done
Attaching to metaflow-service_db_1, metaflow-service_ui_backend_1, metaflow-service_migration_1, metaflow-service_metadata_1
db_1          | The files belonging to this database system will be owned by user "postgres".
db_1          | This user must also own the server process.
db_1          |
db_1          | The database cluster will be initialized with locale "en_US.utf8".
db_1          | The default database encoding has accordingly been set to "UTF8".
db_1          | The default text search configuration will be set to "english".
db_1          |
db_1          | Data page checksums are disabled.
db_1          |
db_1          | fixing permissions on existing directory /var/lib/postgresql/data ... ok
db_1          | creating subdirectories ... ok
db_1          | selecting default max_connections ... 100
db_1          | selecting default shared_buffers ... 128MB
db_1          | selecting default timezone ... Etc/UTC
db_1          | selecting dynamic shared memory implementation ... posix
db_1          | creating configuration files ... ok
db_1          | running bootstrap script ... ok
db_1          | performing post-bootstrap initialization ... ok
db_1          | syncing data to disk ... ok
db_1          |
db_1          | WARNING: enabling "trust" authentication for local connections
db_1          | You can change this by editing pg_hba.conf or using the option -A, or
db_1          | --auth-local and --auth-host, the next time you run initdb.
db_1          |
db_1          | Success. You can now start the database server using:
db_1          |
db_1          |     pg_ctl -D /var/lib/postgresql/data -l logfile start
db_1          |
db_1          | waiting for server to start....2022-04-07 13:42:03.280 UTC [49] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
db_1          | 2022-04-07 13:42:03.304 UTC [50] LOG:  database system was shut down at 2022-04-07 13:42:02 UTC
db_1          | 2022-04-07 13:42:03.312 UTC [49] LOG:  database system is ready to accept connections
db_1          |  done
db_1          | server started
db_1          |
db_1          | /usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
db_1          |
db_1          | 2022-04-07 13:42:03.602 UTC [49] LOG:  received fast shutdown request
db_1          | waiting for server to shut down....2022-04-07 13:42:03.605 UTC [49] LOG:  aborting any active transactions
db_1          | 2022-04-07 13:42:03.611 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
db_1          | 2022-04-07 13:42:03.611 UTC [51] LOG:  shutting down
db_1          | 2022-04-07 13:42:03.637 UTC [49] LOG:  database system is shut down
db_1          |  done
db_1          | server stopped
db_1          |
db_1          | PostgreSQL init process complete; ready for start up.
db_1          |
db_1          | 2022-04-07 13:42:03.729 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
db_1          | 2022-04-07 13:42:03.729 UTC [1] LOG:  listening on IPv6 address "::", port 5432
db_1          | 2022-04-07 13:42:03.741 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
db_1          | 2022-04-07 13:42:03.767 UTC [68] LOG:  database system was shut down at 2022-04-07 13:42:03 UTC
db_1          | 2022-04-07 13:42:03.776 UTC [1] LOG:  database system is ready to accept connections
metadata_1    | INFO:AsyncPostgresDB:global:Connection established.
metadata_1    |    Pool min: 1 max: 10
metadata_1    |
ui_backend_1  | INFO:AsyncPostgresDB:ui:Setting up notify trigger for flows_v3
ui_backend_1  |    Keys: ['flow_id']
db_1          | 2022-04-07 13:42:06.024 UTC [79] ERROR:  relation "public.flows_v3" does not exist
db_1          | 2022-04-07 13:42:06.024 UTC [79] STATEMENT:
db_1          | 	            CREATE TRIGGER notify_ui_flows_v3 AFTER INSERT OR UPDATE OR DELETE ON public.flows_v3
db_1          | 	                FOR EACH ROW EXECUTE PROCEDURE public.notify_ui_flows_v3();
db_1          |
ui_backend_1  | ERROR:AsyncPostgresDB:ui:Exception occurred
ui_backend_1  | Traceback (most recent call last):
ui_backend_1  |   File "/root/services/data/postgres_async_db.py", line 85, in _init
ui_backend_1  |     await table._init(create_triggers=create_triggers)
ui_backend_1  |   File "/root/services/data/postgres_async_db.py", line 162, in _init
ui_backend_1  |     await PostgresUtils.setup_trigger_notify(db=self.db, table_name=self.table_name, keys=self.trigger_keys)
ui_backend_1  |   File "/root/services/data/postgres_async_db.py", line 433, in setup_trigger_notify
ui_backend_1  |     commands=_commands
ui_backend_1  |   File "/root/services/data/postgres_async_db.py", line 365, in create_trigger_if_missing
ui_backend_1  |     await cur.execute(command)
ui_backend_1  |   File "/usr/local/lib/python3.7/site-packages/aiopg/connection.py", line 426, in execute
ui_backend_1  |     await self._conn._poll(waiter, timeout)
ui_backend_1  |   File "/usr/local/lib/python3.7/site-packages/aiopg/connection.py", line 881, in _poll
ui_backend_1  |     await asyncio.wait_for(self._waiter, timeout)
ui_backend_1  |   File "/usr/local/lib/python3.7/asyncio/tasks.py", line 442, in wait_for
ui_backend_1  |     return fut.result()
ui_backend_1  |   File "/usr/local/lib/python3.7/site-packages/aiopg/connection.py", line 788, in _ready
ui_backend_1  |     state = self._conn.poll()
ui_backend_1  | psycopg2.errors.UndefinedTable: relation "public.flows_v3" does not exist
ui_backend_1  |
ui_backend_1  | /usr/local/lib/python3.7/site-packages/aiopg/pool.py:479: ResourceWarning: Unclosed 1 connections in <aiopg.pool.Pool object at 0x7f98f418fad0>
ui_backend_1  |   f"Unclosed {left} connections in {self!r}", ResourceWarning
ui_backend_1  | INFO:AsyncPostgresDB:ui:Setting up notify trigger for flows_v3
ui_backend_1  |    Keys: ['flow_id']
db_1          | 2022-04-07 13:42:07.061 UTC [80] ERROR:  relation "public.flows_v3" does not exist
db_1          | 2022-04-07 13:42:07.061 UTC [80] STATEMENT:
db_1          | 	            CREATE TRIGGER notify_ui_flows_v3 AFTER INSERT OR UPDATE OR DELETE ON public.flows_v3
db_1          | 	                FOR EACH ROW EXECUTE PROCEDURE public.notify_ui_flows_v3();
db_1          |
ui_backend_1  | ERROR:AsyncPostgresDB:ui:Exception occurred
ui_backend_1  | Traceback (most recent call last):
ui_backend_1  |   File "/root/services/data/postgres_async_db.py", line 85, in _init
ui_backend_1  |     await table._init(create_triggers=create_triggers)
ui_backend_1  |   File "/root/services/data/postgres_async_db.py", line 162, in _init
ui_backend_1  |     await PostgresUtils.setup_trigger_notify(db=self.db, table_name=self.table_name, keys=self.trigger_keys)
ui_backend_1  |   File "/root/services/data/postgres_async_db.py", line 433, in setup_trigger_notify
ui_backend_1  |     commands=_commands
ui_backend_1  |   File "/root/services/data/postgres_async_db.py", line 365, in create_trigger_if_missing
ui_backend_1  |     await cur.execute(command)
ui_backend_1  |   File "/usr/local/lib/python3.7/site-packages/aiopg/connection.py", line 426, in execute
ui_backend_1  |     await self._conn._poll(waiter, timeout)
ui_backend_1  |   File "/usr/local/lib/python3.7/site-packages/aiopg/connection.py", line 881, in _poll
ui_backend_1  |     await asyncio.wait_for(self._waiter, timeout)
ui_backend_1  |   File "/usr/local/lib/python3.7/asyncio/tasks.py", line 442, in wait_for
ui_backend_1  |     return fut.result()
ui_backend_1  |   File "/usr/local/lib/python3.7/site-packages/aiopg/connection.py", line 788, in _ready
ui_backend_1  |     state = self._conn.poll()
ui_backend_1  | psycopg2.errors.UndefinedTable: relation "public.flows_v3" does not exist
ui_backend_1  |
ui_backend_1  | /usr/local/lib/python3.7/site-packages/aiopg/pool.py:479: ResourceWarning: Unclosed 1 connections in <aiopg.pool.Pool object at 0x7f98f418fe50>
ui_backend_1  |   f"Unclosed {left} connections in {self!r}", ResourceWarning
ui_backend_1  | INFO:AsyncPostgresDB:ui:Setting up notify trigger for flows_v3
ui_backend_1  |    Keys: ['flow_id']
db_1          | 2022-04-07 13:42:08.083 UTC [81] ERROR:  relation "public.flows_v3" does not exist
db_1          | 2022-04-07 13:42:08.083 UTC [81] STATEMENT:
db_1          | 	            CREATE TRIGGER notify_ui_flows_v3 AFTER INSERT OR UPDATE OR DELETE ON public.flows_v3
db_1          | 	                FOR EACH ROW EXECUTE PROCEDURE public.notify_ui_flows_v3();
db_1          |
ui_backend_1  | ERROR:AsyncPostgresDB:ui:Exception occurred
ui_backend_1  | Traceback (most recent call last):
ui_backend_1  |   File "/root/services/data/postgres_async_db.py", line 85, in _init
ui_backend_1  |     await table._init(create_triggers=create_triggers)
ui_backend_1  |   File "/root/services/data/postgres_async_db.py", line 162, in _init
ui_backend_1  |     await PostgresUtils.setup_trigger_notify(db=self.db, table_name=self.table_name, keys=self.trigger_keys)
ui_backend_1  |   File "/root/services/data/postgres_async_db.py", line 433, in setup_trigger_notify
ui_backend_1  |     commands=_commands
ui_backend_1  |   File "/root/services/data/postgres_async_db.py", line 365, in create_trigger_if_missing
ui_backend_1  |     await cur.execute(command)
ui_backend_1  |   File "/usr/local/lib/python3.7/site-packages/aiopg/connection.py", line 426, in execute
ui_backend_1  |     await self._conn._poll(waiter, timeout)
ui_backend_1  |   File "/usr/local/lib/python3.7/site-packages/aiopg/connection.py", line 881, in _poll
ui_backend_1  |     await asyncio.wait_for(self._waiter, timeout)
ui_backend_1  |   File "/usr/local/lib/python3.7/asyncio/tasks.py", line 442, in wait_for
ui_backend_1  |     return fut.result()
ui_backend_1  |   File "/usr/local/lib/python3.7/site-packages/aiopg/connection.py", line 788, in _ready
ui_backend_1  |     state = self._conn.poll()
ui_backend_1  | psycopg2.errors.UndefinedTable: relation "public.flows_v3" does not exist
ui_backend_1  |
ui_backend_1  | Traceback (most recent call last):
ui_backend_1  |   File "/usr/local/bin/ui_backend_service", line 33, in <module>
ui_backend_1  |     sys.exit(load_entry_point('metadata-service', 'console_scripts', 'ui_backend_service')())
ui_backend_1  |   File "/root/services/ui_backend_service/ui_server.py", line 129, in main
ui_backend_1  |     the_app = app(loop, DBConfiguration())
ui_backend_1  |   File "/root/services/ui_backend_service/ui_server.py", line 54, in app
ui_backend_1  |     loop.run_until_complete(async_db._init(db_conf=db_conf, create_triggers=DB_TRIGGER_CREATE))
ui_backend_1  |   File "/usr/local/lib/python3.7/asyncio/base_events.py", line 587, in run_until_complete
ui_backend_1  |     return future.result()
ui_backend_1  |   File "/root/services/data/postgres_async_db.py", line 97, in _init
ui_backend_1  |     raise e
ui_backend_1  |   File "/root/services/data/postgres_async_db.py", line 85, in _init
ui_backend_1  |     await table._init(create_triggers=create_triggers)
ui_backend_1  |   File "/root/services/data/postgres_async_db.py", line 162, in _init
ui_backend_1  |     await PostgresUtils.setup_trigger_notify(db=self.db, table_name=self.table_name, keys=self.trigger_keys)
ui_backend_1  |   File "/root/services/data/postgres_async_db.py", line 433, in setup_trigger_notify
ui_backend_1  |     commands=_commands
ui_backend_1  |   File "/root/services/data/postgres_async_db.py", line 365, in create_trigger_if_missing
ui_backend_1  |     await cur.execute(command)
ui_backend_1  |   File "/usr/local/lib/python3.7/site-packages/aiopg/connection.py", line 426, in execute
ui_backend_1  |     await self._conn._poll(waiter, timeout)
ui_backend_1  |   File "/usr/local/lib/python3.7/site-packages/aiopg/connection.py", line 881, in _poll
ui_backend_1  |     await asyncio.wait_for(self._waiter, timeout)
ui_backend_1  |   File "/usr/local/lib/python3.7/asyncio/tasks.py", line 442, in wait_for
ui_backend_1  |     return fut.result()
ui_backend_1  |   File "/usr/local/lib/python3.7/site-packages/aiopg/connection.py", line 788, in _ready
ui_backend_1  |     state = self._conn.poll()
ui_backend_1  | psycopg2.errors.UndefinedTable: relation "public.flows_v3" does not exist
ui_backend_1  |
metaflow-service_ui_backend_1 exited with code 1
```

**Comment by oavdeev** [comment]
Can you try the updated docker-compose file from #309? You'll likely need to run `docker-compose -f docker-compose.development.yml build` before running `docker-compose -f docker-compose.development.yml up` again.

**Comment by gaborvecsei** [comment]
@oavdeev seems like you solved it, no more `public.flows_v3` error. Thank you! 🙂  

But I have other issues. I'll comment here, but tell me if I should open a new issue.

On the UI I have the following error when opening up `localhost:3000`:
<img width="1426" alt="image" src="https://user-images.githubusercontent.com/18753533/162408581-cc10b277-8ae2-4352-97a8-f65d0a4251e9.png">

And when I tried opening `localhost:8083/api` this is what I got:
```
[Errno 2] No such file or directory: '/root/services/ui_backend_service/ui/index.html'
```
